src/main/java/chess/view/InputView.java

```diff
@@ -79,6 +89,26 @@ private static Command renderToCommand(String input) {
         }
     }
 
+    private enum PromotionPiece {
+        QUEEN("q"),
+        BISHOP("b"),
+        KNIGHT("n"),
+        ROOK("r");
+
+        private final String command;
+
+        PromotionPiece(String command) {
+            this.command = command;
+        }
+
+        private static void validate(String input) {
+            Arrays.stream(values())
+                    .filter(value -> value.command.equals(input))
+                    .findAny()
+                    .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 기물입니다."));
+        }
```

Comment:
[반환값을 무시하지 말라](https://wiki.sei.cmu.edu/confluence/display/java/EXP00-J.+Do+not+ignore+values+returned+by+methods)는 권장사항을 따라보면 어떨까요?

```suggestion
        private static void validate(String input) {
            Optional<PromotionPiece> optionalPromotionPiece = Arrays.stream(values())
                    .filter(value -> value.command.equals(input))
                    .findAny();
            
            if (optionalPromotionPiece.isEmpty()) {
                throw new IllegalArgumentException("존재하지 않는 기물입니다.");
            }
        }
```